app/apps/handlers.js

```diff
@@ -1,12 +1,22 @@
 const { App, Bucket, User } = require('../models');
-const { repos } = require('../config');
+const { Repo } = require('../models/github');
+const config = require('../config');
+const github = require('../api_clients/github');
+
 
 exports.new = (req, res, next) => {
-  Bucket.list()
-    .then((buckets) => {
+  github.get_access_token(req.user)
```

Comment:
I wonder if we could hide this inside the `github` API client and just call `github.authenticate(user)` which would get the access token and authenticate internally, e.g.

```javascript
github.authenticate(user).
    .then(() => {
      return Promise.all([Repo.list(), Bucket.list()]);
    })
    .then(([repos, buckets]) => {
    // ...
```

Or something like that, but I may miss some reason why you added `github.get_access_token()` in the first place.

Reply:
That is a great idea 👍